File: app/src/main/java/com/selflearning/chemistree/chemistry/inorganic/oxides/OxidesDao.java
package com.selflearning.chemistree.chemistry.inorganic.oxides;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface OxidesDao {
    @Query("SELECT * FROM oxides")
    LiveData<List<Oxides>> getOxides();

    @Insert
    void insertAll(List<Oxides> oxides);
}

File: app/src/main/java/com/selflearning/chemistree/domain/chemistry/inorganic/oxides/OxidesDao.kt
package com.selflearning.chemistree.domain.chemistry.inorganic.oxides

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface OxidesDao {

    @Query("SELECT * FROM oxides")
    fun getAll(): LiveData<List<Oxides>>

    @Insert
    fun insertAll(oxides: List<Oxides?>?)
}